src/tools/wasm-fuzz-types.cpp

```diff
@@ -509,6 +512,95 @@ void Fuzzer::checkInhabitable() {
   }
 }
 
+void Fuzzer::checkRecGroupShapes() {
+  using ShapeHash = std::hash<RecGroupShape>;
+
+  // Collect the groups and order types by index.
+  std::vector<std::vector<HeapType>> groups;
+  std::unordered_map<HeapType, Index> typeIndices;
+  for (auto type : types) {
+    typeIndices.insert({type, typeIndices.size()});
+    if (type.getRecGroupIndex() == 0) {
```

Comment:
What does index 0 mean that is special? I looked at the definition of the function in `wasm-type.h` and don't see an explanation. In fact I'm not sure what the index means - is it some kind of global unique index?

Reply:
`getRecGroupIndex` gives the index of type within its rec group, so whenever it is zero, we know we are starting a new rec group.

Reply:
I will add a comment.

Reply:
Thanks. Please also add a comment to `wasm-types.h`.